src/app/util/attribute-table.cpp

```diff
@@ -546,15 +603,19 @@ EmberAfStatus emAfWriteAttribute(EndpointId endpoint, ClusterId cluster, Attribu
 #endif // BIGENDIAN_CPU
             minR = emberAfCompareValues(minI, data, dataLen, isAttributeSigned);
             maxR = emberAfCompareValues(maxI, data, dataLen, isAttributeSigned);
-            if ((minR == 1) || (maxR == -1))
+            if (((minR == 1) || (maxR == -1)) &&
+                // null value is always in-range for a nullable attribute.
+                (!metadata->IsNullable() || !IsNullValue(data, dataLen, isAttributeSigned)))
             {
                 return EMBER_ZCL_STATUS_INVALID_VALUE;
             }
         }
         else
         {
-            if ((emberAfCompareValues(minv.ptrToDefaultValue, data, dataLen, isAttributeSigned) == 1) ||
-                (emberAfCompareValues(maxv.ptrToDefaultValue, data, dataLen, isAttributeSigned) == -1))
+            if (((emberAfCompareValues(minv.ptrToDefaultValue, data, dataLen, isAttributeSigned) == 1) ||
+                 (emberAfCompareValues(maxv.ptrToDefaultValue, data, dataLen, isAttributeSigned) == -1)) &&
+                // null value is always in-range for a nullable attribute.
+                (!metadata->IsNullable() || !IsNullValue(data, dataLen, isAttributeSigned)))
```

Comment:
Would it be possible to rewrite it a little as it is unclear that we need to duplicate the code ? Something like:
```
bool isOutOfRange = false;
if (dataLen <= 2)
{
   ...
   isOutOfRange = emberAfCompareValues(minI, data, dataLen, isAttributeSigned) == 1 ||
                  emberAfCompareValues(maxI, data, dataLen, isAttributeSigned) == -1);
}
else
{
    isOutOfRange = emberAfCompareValues(minv.ptrToDefaultValue, data, dataLen, isAttributeSigned) == 1 ||
                   emberAfCompareValues(maxv.ptrToDefaultValue, data, dataLen, isAttributeSigned) == -1;
}

if (isOutOfRange && (!metadata.IsNullable() && !IsNullValue(data, dataLen, isAttributeSigned)))
{
    return EMBER_ZCL_STATUS_INVALID_VALUE;
}

Reply:
https://github.com/project-chip/connectedhomeip/pull/13059